Zoom.cpp

```diff
@@ -108,11 +109,31 @@ void VertZoomRange(int iFirst, int iNum, bool* bZoomed, bool bMinimizeOthers, bo
 	int iTotalHeight = rect.bottom;
 	int lastTrackId = iFirst + iNum - (includeEnvelopes ? 1 : 2);
 
+	const int minTrackHeight = SNM_GetIconTheme()->tcp_small_height;
+	const bool obeyHeightLock = NF_IsObeyTrackHeightLockEnabled();
+	
 	if (bMinimizeOthers)
 	{
 		*ConfigVar<int>("vzoom2") = 0;
-		for (int i = 0; i <= GetNumTracks(); i++)
-			GetSetMediaTrackInfo(CSurf_TrackFromID(i, false), "I_HEIGHTOVERRIDE", &g_i0);
+		if (obeyHeightLock) 
+		{
+			for (int i = 0; i <= GetNumTracks(); i++) {
+				MediaTrack* tr = CSurf_TrackFromID(i, false);
+				if (!SWS_IsTrackHeightLocked(tr)) {
+					SetMediaTrackInfo_Value(tr, "I_HEIGHTOVERRIDE", g_i0);
+				}		
+			}
+		}
+		else 
+		{
+			for (int i = 0; i <= GetNumTracks(); i++) {
+				MediaTrack* tr = CSurf_TrackFromID(i, false);
+				// NF: can't use SetMediaTrackInfo_Value(tr, "I_HEIGHTOVERRIDE", 0) here, because it would disable height lock
+				// https://forum.cockos.com/showthread.php?p=2202082#post2202082, #1221
+				SetMediaTrackInfo_Value(tr, "I_HEIGHTOVERRIDE", minTrackHeight);
+			}
+		}
```

Comment:
Somehow (I don't understand how the whole function works, perhaps it happens via action 40112?), setting the height override of non-locked tracks to `minTrackHeight` rather than 0 sometimes prevents them from being minimized:

![Screencap](https://i.imgur.com/NxEr03x.gif)

This fixes it (I also merged the two branches of the `obeyHeightLock` condition to simplify the code):

```suggestion
		for (int i = 0; i <= GetNumTracks(); i++)
		{
			MediaTrack* tr = CSurf_TrackFromID(i, false);
			const bool locked = SWS_IsTrackHeightLocked(tr);

			if (!obeyHeightLock || !locked)
				SetMediaTrackInfo_Value(tr, "I_HEIGHTOVERRIDE", locked ? minTrackHeight : 0);
		}
```